src/GraphQL/Resolver/ImageAssetFieldResolver.php

```
@@ -6,52 +6,50 @@
  */
 namespace EzSystems\EzPlatformGraphQL\GraphQL\Resolver;
 
-use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
-use eZ\Publish\Core\FieldType\ImageAsset\AssetMapper;
-use EzSystems\EzPlatformGraphQL\GraphQL\DataLoader\ContentLoader;
 use EzSystems\EzPlatformGraphQL\GraphQL\Value\Field;
+use Ibexa\GraphQL\GraphQL\Mapper\ImageAssetMapperStrategyInterface;
 
 /**
  * @internal
  */
 class ImageAssetFieldResolver
 {
+    /* @var \Ibexa\Core\FieldType\ImageAsset\ImageAssetMapperStrategyInterface[] */
+    private $strategies;
+
     /**
-     * @var DomainContentResolver
-     */
-    private $domainContentResolver;
-    /**
-     * @var ContentLoader
-     */
-    private $contentLoader;
-    /**
-     * @var AssetMapper
+     * @param iterable<ImageAssetMapperStrategyInterface> $strategies
      */
-    private $assetMapper;
+    public function __construct(iterable $strategies)
+    {
+        foreach ($strategies as $strategy) {
+            if ($strategy instanceof ImageAssetMapperStrategyInterface) {
+                $this->addStrategy($strategy);
+            }
+        }
+    }
 
-    public function __construct(ContentLoader $contentLoader, DomainContentResolver $domainContentResolver, AssetMapper $assetMapper)
+    private function addStrategy(ImageAssetMapperStrategyInterface $strategy): void
     {
-        $this->domainContentResolver = $domainContentResolver;
-        $this->contentLoader = $contentLoader;
-        $this->assetMapper = $assetMapper;
+        $this->strategies[] = $strategy;
     }
 
-    public function resolveDomainImageAssetFieldValue(Field $field)
+    public function resolveDomainImageAssetFieldValue(Field $field): ?Field
```

Maybe instead of changing return type you should throw an exception?

---

What kind of exception should I throw and where? This method doesn't throw any and `null` is a proper return type here.

---

You're right, I overlooked it. :+1: